### DISCUSSION TITLE
.env vars inside virtualenvs

### DISCUSSION THREAD

**Discussion by startakovsky** [python-poetry/poetry#8147] [General] [answered]
Hi, I am coming from using pipenv enjoyed that pipenv automatically instantiated assignments in the file `.env`. Does poetry offer any analog within the virtual environment?  I am familiar with python-dotenv but would prefer avoiding that in my library.

**Comment by finswimmer** [comment, marked as answer]
Poetry does not support this out-of-the-box, because we believe this is not a task for a development tool. Instead either the application should be responsible to load the env vars e.g. by python-dotenv or it's part of the installation process of the application.

However, poetry plugins exists, e.g.  https://pypi.org/project/poetry-plugin-dotenv/

**Reply by mpeteuil** [reply]
There is also https://pypi.org/project/poetry-dotenv-plugin/ [^1] which relies on `python-dotenv` to do much of the work[^2], [making the plugin code very simple](https://github.com/mpeteuil/poetry-dotenv-plugin/blob/22eaf2d/poetry_dotenv_plugin/dotenv_plugin.py#L37). Either should solve this issue though.

[^1]: It's confusing that there are two with essentially the same name. This one was created before there was a naming convention to follow.
[^2]: you won't need `python-dotenv` as a dependency in your project
